shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureWrapper.java

```diff
@@ -53,16 +55,33 @@ public void release() {
   @SuppressWarnings("unused")
   public void attachToGLContext(int texName) {
     synchronized (this) {
+      // When the rasterizer tasks run on a different thread, the GrContext is re-created.
+      // This causes the texture to be in an uninitialized state.
+      // This should *not* be an issue once platform views are always rendered as TextureLayers
+      // since thread merging will be always disabled on Android.
+      // For more see: AndroidExternalTextureGL::OnGrContextCreated in
+      // android_external_texture_gl.cc, and
+      // https://github.com/flutter/flutter/issues/98155
+      if (attached) {
+        surfaceTexture.detachFromGLContext();
+        attached = false;
+      }
       if (!released) {
         surfaceTexture.attachToGLContext(texName);
+        attached = true;
       }
     }
   }
 
   // Called by native.
   @SuppressWarnings("unused")
   public void detachFromGLContext() {
-    surfaceTexture.detachFromGLContext();
+    synchronized (this) {
+      if (!released) {
```

Comment:
Should this be gated on `attached` instead of `!released`?

Reply:
yes, sorry. I made this a little safer in https://github.com/flutter/engine/pull/31390/commits/afd880dbf7de45f55c15112dfaf46e7a8d162b26